File: JoyReactor.Android/Widget/ColorSwipeRefreshLayout.cs
using Android.Content;
using Android.Support.V4.Widget;
using Android.Util;

namespace JoyReactor.Android.Widget
{
    public class ColorSwipeRefreshLayout : SwipeRefreshLayout
    {
        bool initialized;

        public ColorSwipeRefreshLayout(Context context, IAttributeSet attrs)
            : base(context, attrs)
        {
            SetColorScheme(
                global::Android.Resource.Color.HoloBlueBright, 
                global::Android.Resource.Color.HoloGreenLight,
                global::Android.Resource.Color.HoloOrangeLight, 
                global::Android.Resource.Color.HoloRedLight);
        }

        protected override void OnLayout(bool changed, int left, int top, int right, int bottom)
        {
            base.OnLayout(changed, left, top, right, bottom);
            FixFirstShow();
        }

        void FixFirstShow()
        {
            if (!initialized)
            {
                initialized = true;
                Refreshing = !Refreshing;
                Refreshing = !Refreshing;
            }
        }
    }
}

File: android/src/main/kotlin/y2k/joyreactor/widget/ColorSwipeRefreshLayout.kt
package y2k.joyreactor.widget

import android.content.Context
import android.support.v4.widget.SwipeRefreshLayout
import android.util.AttributeSet
import y2k.joyreactor.R

/**
 * Created by y2k on 9/26/15.
 */
class ColorSwipeRefreshLayout(context: Context, attrs: AttributeSet) : SwipeRefreshLayout(context, attrs) {

    init {
        setColorSchemeResources(R.color.primary, R.color.primary_dark);
    }
}